### ISSUE TITLE
Separate out implementations into launcher

### ISSUE THREAD

**Issue by kevzhao2** [NyxStudios/Orion#73]
We need to separate out the implementations of the interfaces such as `IItem`, etc. into the launcher.

This allows Orion.Core to _only_ be a set of APIs, and the concrete, Terraria-facing stuff gets put into the launcher.

**Comment by hakusaro** [comment]
Based on that last sentence, Launcher seems like a really odd choice of words. Can you elaborate more on that?
On Fri, Jun 19, 2020 at 11:39 PM GitHub <notifications@github.com> wrote:
“We need to separate out the implementations of the interfaces such as IItem, etc. into the launcher.

This allows Orion.Core to only be a set of APIs, and the concrete, Terraria-facing stuff gets put into the launcher.”

**Comment by kevzhao2** [comment]
It's the entire set of things which will be required to launch an Orion instance.

So this means that implementations, like `OrionItem`, will be done in the "launcher". (There'll be corresponding tests in that, as well). Not sure if there's a better way to name this?

Basically, I don't want to expose OTAPI/Terraria from Orion.Core.

**Comment by hakusaro** [comment]
I’m not sure, but I’ll tell you that “Launcher” makes me think of like the Minecraft launcher and not a set of APIs, if that makes sense.
On Fri, Jun 19, 2020 at 11:53 PM GitHub <notifications@github.com> wrote:
“It's the entire set of things which will be required to launch an Orion instance.

So this means that implementations, like OrionItem, will be done in the "launcher". (There'll be corresponding tests in that, as well). Not sure if there's a better way to name this?

Basically, I don't want to expose OTAPI/Terraria from Orion.Core.”

**Comment by kevzhao2** [comment]
I'll go with the launcher terminology for now. It should be fairly easy to rename in the future.

For the record, the launcher doesn't provide any APIs for consumption.
